demos/vc_issuer/tests/issue_credential.rs

```diff
@@ -353,12 +362,34 @@ fn should_return_derivation_origin() {
     let canister_id = install_canister(&env, VC_ISSUER_WASM.clone());
     let frontend_hostname = format!("https://{}.icp0.io", canister_id.to_text());
     let req = DerivationOriginRequest { frontend_hostname };
-    let response = api::derivation_origin(&env, canister_id, principal_1(), &req)
+    let response = api::derivation_origin(&env, canister_id, &req)
         .expect("API call failed")
         .expect("derivation_origin error");
     assert_eq!(response.origin, req.frontend_hostname);
 }
 
+#[test]
+fn should_set_derivation_origin() {
+    let env = env();
+    let canister_id = install_canister(&env, VC_ISSUER_WASM.clone());
+    let frontend_hostname = "frontend_hostname.com".to_string();
+    let derivation_origin = "https://derivation.origin";
+
+    api::set_derivation_origin(
+        &env,
+        canister_id,
+        frontend_hostname.as_str(),
+        derivation_origin,
+    )
+    .expect("failed to set derivation_origin");
+
+    let req = DerivationOriginRequest { frontend_hostname };
+    let response = api::derivation_origin(&env, canister_id, &req)
```

Comment:
Could you check this before calling `set_derivation_origin` to confirm that it has indeed changed?